app/src/lib/release-notes.ts

```diff
@@ -0,0 +1,105 @@
+const ordinal: (i: number) => string = require('ordinal')
+
+import {
+  ReleaseMetadata,
+  ReleaseNote,
+  ReleaseSummary,
+} from '../models/release-notes'
+
+const itemEntryRe = /^\[(new|fixed|improved|removed|added|pretext)\]\s(.*)/i
+
+const monthNames = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+]
+
+function formatDate(date: Date) {
+  const day = date.getDate()
+  const dayWithSuffix = ordinal(day)
+  const monthIndex = date.getMonth()
+  const year = date.getFullYear()
+
+  return `${monthNames[monthIndex]} ${dayWithSuffix} ${year}`
+}
+
+export function parseReleaseEntries(
+  notes: ReadonlyArray<string>
+): ReadonlyArray<ReleaseNote> {
+  const entries = new Array<ReleaseNote>()
+
+  for (const note of notes) {
+    const text = note.trim()
+    const match = itemEntryRe.exec(text)
+    if (match === null) {
+      continue
+    }
+    const kind = match[1].toLowerCase()
+    const message = match[2]
+
+    switch (kind) {
```

Comment:
could we use the regex at https://github.com/desktop/desktop/pull/3601/files#diff-c6f2f498c0eae789227e5aeddca2c0bbR9 to do this filter for us? i'd like to have one canonical place where we define these note types, if possible

Reply:
also, would prefer an `Array.filter` or an `Array.includes` to a switch case in this situation.

Reply:
This was already using the `itemEntryRe` on line 21 but that dealt with loose strings, but I gave this another pass to clean it up and make it more readable - check out f16be005c and see if this makes it easier to follow